### DISCUSSION TITLE
Fly Trajectories on User Command

### DISCUSSION THREAD

**Discussion by sesem738** [bitcraze/discussions#1369] [Q&A]
Hi everyone,

I'm working on a drone project where I need to achieve the following behavior:

1. User Input: The user selects one of several pre-defined 8th-order polynomial spline trajectories.
2. Trajectory Execution: The drone executes the chosen trajectory.
3. Return to Origin: After completing the trajectory, the drone automatically returns to its starting position.
4. Hover and Wait: The drone hovers at the origin, waiting for the next user input command.

Essentially, I want to create a loop where the drone hovers until a new command is given, then follows the selected trajectory, and finally returns to the hover state. What would be the best way to execute something like this? I plan on using the Crazyflie with the Flowdeck.

**Comment by gemenerik** [comment]
Based on your requirements, the easiest implementation would be to use the Crazyflie Python library. For trajectory execution you can look at [this example](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/autonomous_sequence_high_level.py). You can design a simple state machine that makes the drone hover at start by default, and with the Python native [input function](https://docs.python.org/3/library/functions.html#input) get users to select a trajectory. Hope that gets you started.
